source/geometry_model/initial_topography_model/ascii_data.cc

```diff
@@ -0,0 +1,182 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/geometry_model/initial_topography_model/ascii_data.h>
+#include <aspect/geometry_model/box.h>
+#include <aspect/geometry_model/sphere.h>
+#include <aspect/geometry_model/spherical_shell.h>
+#include <aspect/geometry_model/chunk.h>
+
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/std_cxx11/array.h>
+
+
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    AsciiData<dim>::AsciiData ()
+      :
+      surface_boundary_id(1)
+    {}
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::initialize ()
+    {
+      // Get the symbolic names used by the current geometry
+      const std::map<std::string,types::boundary_id> boundary_names = this->get_geometry_model().get_symbolic_boundary_names_map();
+
+      // Get the boundary id number for the boundary that is the surface of the current geometry
+      for (std::map<std::string,types::boundary_id>::const_iterator i = boundary_names.begin();
+           i != boundary_names.end();
+           i ++)
+        {
+          if (i->first == "surface" || i->first == "top" || i->first == "outer")
```

Comment:
This may be correct right now, but it is going to lead to some sort of problem at one point or other. Can you qualify this with the geometry model somehow? E.g.
```
  if ((geometry model is box) && (i->first == "top")
      ||
      other options
```